src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc

```diff
@@ -53,11 +56,37 @@ bool RCRPCPlugin::Init(
                                           hmi_capabilities,
                                           policy_handler,
                                           *resource_allocation_manager_,
-                                          *interior_data_cache_));
+                                          *interior_data_cache_,
+                                          *this));
+
   interior_data_cache_.reset(new InteriorDataCacheImpl());
+
+  const auto& settings = app_manager.get_settings();
+  const auto frequency = settings.get_interior_vehicle_data_frequency();
+  max_requests_count = frequency.first;
+
+  exit_timer_signal_ = std::promise<void>();
+
+  timer_thread = std::thread(
```

Comment:
@ZhdanovP pease use for timer existing class Timer. Keep consistency in using utils with the project. 
The example you can find in Resumption : 

```
save_persistent_data_timer_("RsmCtrlPercist",
                                  new timer::TimerTaskImpl<ResumeCtrlImpl>(
                                      this, &ResumeCtrlImpl::SaveDataOnTimer))

...

void ResumeCtrlImpl::SaveDataOnTimer() {

```